### DISCUSSION TITLE
How to apply conditional CSS Module Classes with Tailwind CSS

### DISCUSSION THREAD

**Discussion by Heet-Bhalodiya** [tailwindlabs/tailwindcss#12589] [Help]
I am using `tailwind:3.3.5` and have encountered a query regarding the conditional application of CSS module classes based on Tailwind's screen size utility. I am seeking guidance on how to effectively render module file classes conditionally using Tailwind CSS screen sizes.

Here is a snippet of my code for reference:

```tsx
<div
  className={classnames("flex flex-col gap-5", {
    [styles.rightWrapper]: isBelowMdScreen && !isBelowSmScreen,
    "is-[100%]": isAboveMdScreen || isBelowSmScreen,
  })}
>
  <div>
    <Typography variant="h4">Login</Typography>
    <Typography>
      Please sign-in to your account and start the adventure
    </Typography>
  </div>
  <div>
    <TextField label="Email" />
    <TextField label="Password" />
  </div>
</div>
```

And also how can I pass MUI props conditionally while using tailwind screen?

Here is a snippet of my code for reference:

```tsx
<Button
  variant='contained'
  {...(isBelowSmScreen && { fullWidth: true })}
>
```

**Comment by alex-krasikau** [comment]
Hey @Heet-Bhalodiya!

The code snippets generally look correct. Could you provide examples to help me better understand the issue?

As for the second one, it can be simplified as:

```jsx
<Button variant='contained' fullWidth={isBelowSmScreen} />
```

**Reply by Heet-Bhalodiya** [reply]
Currently, the component relies on the `useMediaQuery` function to determine screen size breakpoints, and I believe that switching to Tailwind screen classes (such as `sm`, `md`, `lg`, etc.) would be a more efficient approach. 

```tsx
const Login = () => {

  const isAboveMdScreen = useMediaQuery((theme: Theme) => theme.breakpoints.up("md"));
  const isBelowMdScreen = useMediaQuery((theme: Theme) =>theme.breakpoints.down("md"));
  const isBelowSmScreen = useMediaQuery((theme: Theme) =>theme.breakpoints.down("sm"));
  ...

  return (
    <div
      className={classnames("flex flex-col gap-5", {
        [styles.rightWrapper]: isBelowMdScreen && !isBelowSmScreen,
        "is-[100%]": isAboveMdScreen || isBelowSmScreen,
      })}
    >
      <div>
        <Typography variant="h4">Login</Typography>
        <Typography>
          Please sign-in to your account and start the adventure
        </Typography>
      </div>
      <div>
        <TextField label="Email" />
        <TextField label="Password" />
      </div>
    </div>
  );
};

export default Login
```


styles.module.css file: 

```css
.rightWrapper{
	height: '480px',
	border-bottom: '1px solid var(--mui-palette-divider)
}

**Reply by alex-krasikau** [reply]
You can enable the use of Tailwind Modifiers by wrapping your class definition with the `@layer` directive. This allows you to switch to regular CSS instead of CSS Modules for simplicity. Here's an example: [Tailwind Play](https://play.tailwindcss.com/6HF4IhfcOA).

Regarding setting attributes based on Tailwind styling, it's not possible except by using duplicated definitions. For example, you can use the `hidden md:block` class on the first element and `md:hidden` on the second element. As for `fullWidth`, you can directly set `w-full` to the element for the desired behavior.

**Reply by Heet-Bhalodiya** [reply]
Is there any way to achieve this without using `@layer`? I mean in css module file because I want to write styles for specific page not global.

Thanks for the help !!

**Reply by alex-krasikau** [reply]
No, Tailwind CSS is a build-time tool that analyzes your source code and generates a single CSS file based on the content of the source files. For a simple case like this, you could consider removing CSS modules and either inline the styles or extract them into a separate component.

In my projects, I rarely need to write custom CSS. Most of the time, I work with keyframes and the latest browser APIs.
